src/main/java/org/radarcns/hdfs/util/commandline/CommandLineArgs.java

```diff
@@ -38,6 +41,18 @@
     @Parameter(names = { "-h", "--help"}, help = true, description = "Display the usage of the program with available options.")
     public boolean help;
 
-    @Parameter(names = { "--no-stage"}, description = "Do not stage output files into a temporary directory before moving them to the data directory. This increases performance but may leave corrupted data files.")
-    public boolean noStage = false;
+    @Parameter(names = { "--path-factory" }, description = "Factory to create path names with")
+    public String pathFactory;
+
```

Comment:
Yes, it was removed... Basically the FileCache was getting too complex with multiple branches. Also, I wasn't sure that if we implement an S3 backend, appending to a S3 bucket file would work as intended. If absolutely needs to be in, I can move it back. But as stated in the README, it will have chances of corruption.

Reply:
I understand the disadvantages but as the volume of data increases, the app will start to fall back behind the current data and finally be not able to process records if retention is set on HDFS. It would be good to have that option in case we need the offsets to catchup (in trade for few corrupt files if any) and then turn this off again to ensure there are as little corrupt files as possible.

Reply:
I don't see how that would be, since we use 1 hour file intervals, it shouldn't matter how large the backlog is. Once a file cache is open, using a temporary storage doesn't affect conversion speed.

I've done two things to potentially increase performance:
1. operations are now parallelised, meaning multiple topics may be processed at once. Use with `--num-threads` flag.
2. The number of open files in the file cache store is now configurable with `--cache-size`. If it turns out that there are a lot of cache misses, you could try to give the process more memory and increase the cache size.